plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/providers/vfs/model/VFSDirectory.java

```diff
@@ -116,9 +116,11 @@ public int hashCode() {
     }
 
     VFSDirectory compare = (VFSDirectory) obj;
+    // This comparison depends on `getProvider()` to always return a hardcoded value
     return compare.getProvider().equals( getProvider() )
-      && ( ( compare.getConnection() == null && getConnection() == null ) || compare.getConnection()
-      .equals( getConnection() ) )
-      && ( ( compare.getPath() == null && getPath() == null ) || compare.getPath().equals( getPath() ) );
+      && ( ( compare.getConnection() == null && getConnection() == null )
```

Comment:
At first glance hard to read. Just a suggestion. no change since logically will result in the same.

 could also use apache's StringUtils library to replace lines 121 - 124
`StringUtils.equals(compare.getConnection(), getConnection()) &&  StringUtils.equals(compare.getPath(), getPath()) `

https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#equals-java.lang.CharSequence-java.lang.CharSequence-

I'm assuming the apache library is already a dependency.